clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -356,6 +357,23 @@ protected void onJoinComplete(int generation,
         if (assignor == null)
             throw new IllegalStateException("Coordinator selected invalid assignment protocol: " + assignmentStrategy);
 
+        log.debug("{} assignor selected by the group coordinator", assignor);
+
+        List<RebalanceProtocol> assignorSupportedProtocols = assignor.supportedProtocols();
+        Collections.sort(assignorSupportedProtocols);
+        RebalanceProtocol newProtocol = assignorSupportedProtocols.get(assignorSupportedProtocols.size() - 1);
+        if (newProtocol.id() < protocol.id()) {
+            log.error("Latest commonly supported rebalance protocol is {} which is lower than the current commonly supported rebalance protocol {}. "
```

Comment:
I think a more common error scenario is when a new member joining a group with ONLY old assignor in the configs, while all the existing members in the current generation have both old and new assignors. In this case the old would have to be selected.

So in the error message, we should also clarify this case, e.g. saying something that "once a new protocol is selected, even though everyone still supports the old protocol it cannot be downgraded anymore --- even if a new member with only the old protocol joins the group"